server/threads/routes.ts

```diff
@@ -608,4 +609,54 @@ router.delete(
   }
 );
 
+/**
+ * @openapi
+ * /threads/{thread_id}/delete:
+ *   delete:
+ *     summary: Hard delete of thread in entirety
+ *     operationId: deleteThread
+ *     description: Deletes selected thread from database
+ *     tags:
+ *       - /threads/
+ *     parameters:
+ *       - name: thread_id
+ *         in: path
+ *         description: The id of the thread to fetch.
+ *         required: true
+ *         schema:
+ *           type: string
+ *     responses:
+ *       500:
+ *         description: Internal Server Error
+ *       401:
+ *         $ref: "#/components/responses/ensureLoggedIn401"
+ *       403:
+ *         $ref: "#/components/responses/ensureThreadAccess403"
+ *       404:
+ *         $ref: "#/components/responses/threadNotFound404"
+ */
+
+router.delete(
+  "/:thread_id",
+  ensureLoggedIn,
+  ensureThreadAccess,
+  // ensureModPermissions? at some point
```

Comment:
I would write a guard here to check if we're running in a development environment as part of this PR, just so that you can make it actually throw an Error by default. The risk being that we forget you haven't implemented the check, we accidentally do a release, then anyone can delete threads via API.

I think this should be as easy as doing:

```ts
router.delete(
  "/:thread_id",
  ensureLoggedIn,
  ensureThreadAccess,
  (_, _, next) => {
     if (process.env. NODE_ENV === "development" || process.env.NODE_ENV === "test") {
        next()
     }
     throw new Api500Error("this function is not available in production");
  },
  async (req, res) => {
  // existing code
```

....or something like that! Let me know if it doesn't work for any reason, haven't done this before.

Also feel free to put this somewhere so we can reuse it, if you find (or can create) a nice place!